### DISCUSSION TITLE
How get the domain when user sign in in NextAuth

### DISCUSSION THREAD

**Discussion by Melvynx** [nextauthjs/next-auth#6894] [Help]
Hi, I have created an application that can manage different domains for different schools.

Each school has an associated domain, for example:

`melvynx.codeline.com` => school Jean
`jean.codeline.com` => school Jean

To do this, I need to access the domain when the user signs in.

But in the current callbacks or events, there is never the "request object" to check the domain on sign in.

---

#### What I have tried?

I have tried using the jwt callback, but I cannot find the domain:

```ts
 jwt: async (params) => {
      if (!params.isNewUser) return params.token;

      return params.token;
    },
```

I have also checked the createUser event, but I do not have domain information. Similarly, in the "session callback"...

I think the request is somewhere because, under the hood, the request is caught...

I have tried to "ping" the server just after the request, but I think it's not really a solution. It's much more of a temporary fix.

Can anyone help?

Thank you very much.

**Comment by mhernandezsuk** [comment]
were you able to solve this? I have a similar problem
